### PR TITLE
setup app backend, dev environments, and build tools

### PR DESCRIPTION
**Pull request by tracfren**
*Issue #, if available:*
N/A

*Description of changes:*
* Setup Nest for the back-end framework.
* Use Nest CLI to generate dashboard resource endpoints.
* Configure Nest to use Vite+SWC.
* Configure Nest to use Fastify.
* Setup Turbo Repo to improve development experience of full-stack TS app.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/core/src/dashboards/dashboards.controller.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  @Get()
+  findAll() {
+    return this.dashboardsService.findAll();
+  }
+
+  @Get(":id")
+  findOne(@Param("id") id: string) {
+    return this.dashboardsService.findOne(+id);
```
Comment (chejimmy) [inline apps/core/src/dashboards/dashboards.controller.ts:30]: Why do we want to parse the `id` into a number?

**Review by chejimmy — APPROVED** [review]
🙏 TY for setting this up. I have some minor comments only and proceeding to ship this code. 🚢

File: apps/core/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "core",
```
[the remark this answers is not in the document]
Comment (tracfren) [reply apps/core/package.json:2]: I don't think there's a convention, but Turbo provides many examples (https://github.com/vercel/turbo/tree/main/examples) and it looks like most follow their lead.

File: apps/core/src/dashboards/dashboards.controller.ts (hunk 1 above)
Comment (tracfren) [reply apps/core/src/dashboards/dashboards.controller.ts:30]: This is generated by the NestJS CLI. It looks like there was just an assumption that ID is a number. We can remove this.

**Pull request merged**